### ISSUE TITLE
Does togglz-spring-boot-starter support spring boot 3.0?

### ISSUE THREAD

**Issue by yangxp** [togglz/togglz#910]
spring boot 3.0 using spring framework 6, but togglz-spring-boot-starter using spring framework 5.2.3
https://github.com/togglz/togglz/blob/master/spring-boot/pom.xml

**Comment by bennetelli** [comment]
I did not spent time taking a look at the migration and the steps needed, but it will definitely support spring boot.

**Comment by Jeffrey-Hassan** [comment]
IntelliJ has tools which migrate from javax.* to jakarta.* ... I've run this locally, and along with some other pom.xml changes (changing from some of the jboss spec API depedencies/javax dependencies to jakarta dependencies) I've managed to get togglz to _compile_ with Java17/JakartaEE9/Spring 6/Spring Boot 3, and I can even skip tests and run togglz pretty successfully in a Spring Boot 3 project.

However ... I'm having a whole lot of trouble with this test suite based on Wildfly and Arquillian, unable to get the tests to run successfully (at least locally).  These components are actively developed and do seem to have Jakarta EE9 support (though unfortunately in "Alpha" release only) but just changing all arquillian/wildfly dependencies to latest unfortunately doesn't seem to do the trick, still getting errors looking for javax.servlet classes.  Continuing to work on this, hopefully will come up with something.

**Comment by OleksandrShkurat** [comment]
Is there any chance to see this issue closed by a pull-request with jakarta namespace adaptation?

**Comment by MediaMarco** [comment]
I created a pull request with spring boot 3, spring 6 and jakarta namespace. Three modules currently don't work: cassandra, cdi and shiro. Maybe togglz can have a 4.x.x branch of togglz or maintain the 3.x.x version in a 3.x.x branch and use the main branch for a jakarta-compatible version?

**Comment by OleksandrShkurat** [comment]
It would be greate if more people took part in code review process for #917 
Especially those gyus, who potentially can merge and then release it.
@bennetelli

**Comment by bennetelli** [comment]
Hi there,

there is a first milestone released. 4.0,0-M1. Right now the dynamoDB module as well as the cassandra one are missing. 
I will work on this and release a 4.0.0 (without -M) when everything is finished.

**Comment by bennetelli** [comment]
update: so far I did not have much time to test the milestone and we are working on adding and updating cassandra and dynamodb.

**Comment by bennetelli** [comment]
update. there are some issues with the togglz console. working on it.

**Comment by Harmelodic** [comment]
> update: so far I did not have much time to test the milestone and we are working on adding and updating cassandra and dynamodb.

Just for anyone looking at this project. The cassandra module was dropped:
- https://github.com/togglz/togglz/pull/968

So... just DynamoDB to do(?)

**Comment by Harmelodic** [comment]
and for dynamodb, I believe it's just a case of moving from:
```
com.amazonaws:aws-java-sdk-dynamodb
```
to
```
software.amazon.awssdk:dynamodb
```
and then ensuring all the tests continue to pass, etc.

**Comment by Harmelodic** [comment]
As per comments on this PR:

- https://github.com/togglz/togglz/pull/917

Dynamo DB needs to be migrated to AWS SDK for Java 2.x

I've made a PR for that:

- https://github.com/togglz/togglz/pull/978

**Comment by bennetelli** [comment]
I just released togglz 4.0.0-M2 which contains the current state of the master branch

**Comment by bennetelli** [comment]
> Hi @bennetelli , Just in case I can help, what is the trouble you're having with the console?

adding this line to your Spring Boot Application (that class including the main method)

@Import({TogglzAutoConfiguration.class, TogglzEndpointAutoConfiguration.class, })

fixes the missing console. In case you had this property in your application.properties: togglz.console.use-management-port=true, 
then you can delete it.

**Comment by bennetelli** [comment]
there is a new version 4.0.0-M3 published to maven central.

I will work on the PR which restructures the spring boot starters so they will be part of togglz 4.0.0 as well.

**Comment by bennetelli** [comment]
I am currently fighting with the Spring Boot actuator endpoints. Could be that this is an issue with my configuration, but could also be that this one is a problem with the current milestone release / with the migration of spring boot from 2.7 to 3.

please raise an issue if you find something in the current milestone so we can fix it before releasing togglz-4.0.0 :) thanks in advance and have a nice weekend :)

**Comment by Harmelodic** [comment]
Feedback on milestone:

TL;DR: It's fine.

Long answer:
Upgraded a Spring Boot 2 application to Spring Boot 3, with the `togglz-*:4.0.0-M3` milestone release.

I was using `togglz-junit5:2.8.0` as part of the testing, rather than `togglz-junit:4.0.0-M3`.

Since `togglz-junit5:2.8.0` uses `togglz-core:2.8.0` and the way the dependencies are setup within Maven for my application, meant that when I ran the tests, classes from `togglz-core:2.8.0` were being used instead of from `togglz-core:4.0.0-M3`.

I refactored to just use `togglz-junit:4.0.0-M3` and things seem to work fine.

Though, it'd be nice to have a flag on [`togglz-junit5` in mvnrepository.com](https://mvnrepository.com/artifact/org.togglz/togglz-junit5) to show that people should switch to just `togglz-junit`.

**Comment by bennetelli** [comment]
I will write a blog post containing all the changes in togglz 4. Will also mention the junit dependency there.

**Comment by bennetelli** [comment]
you could fix the merge conflicts with the Spring Boot starters in #888 if you like :) 

everything else is working as expected and done already

**Comment by Harmelodic** [comment]
Seeing as I can't fix the merge conflicts in someone else's repo, I just made a new PR by applying #888 diff, and then refactoring it to use Spring Boot 3 / Jakarta / Spring 6 stuff: #989

**Comment by Harmelodic** [comment]
Any update on when a `4.0.0` can be made?

We're on the brink of just refactoring out Togglz from our projects because it's blocking us from bumping to Spring Boot 3.

**Comment by Harmelodic** [comment]
PR https://github.com/togglz/togglz/pull/888 got merged! 🎉 
PR https://github.com/togglz/togglz/pull/989 got closed! 🥳

`4.0.0` soooooon??? 👀

**Comment by bennetelli** [comment]
currently trying to publish the release

**Comment by bennetelli** [comment]
@Harmelodic GitHub Action says it has been released. But I don't see it in maven central yet. Should be there within the next minutes I guess

**Comment by bennetelli** [comment]
ps: I need to write a blog post about the release and change the release notes so everyone knows what has been changed and knows the breaking changes. Will do that on Tuesday since it's a bank holiday on Monday.

**Comment by bennetelli** [comment]
I can see it on oss sonatype already.

**Comment by bennetelli** [comment]
It has been released already. So I would like to close this issue. Thanks everyone for attending, helping and all the discussions :)

**Comment by kevinstembridge** [comment]
Fantastic work Bennet. Thank you.
You mentioned a blog. Where do you post messages for Togglz?
